test/built-ins/Temporal/PlainMonthDay/prototype/with/basic.js

```diff
@@ -13,11 +13,12 @@ const md = Temporal.PlainMonthDay.from("01-15");
 TemporalHelpers.assertPlainMonthDay(md.with({ day: 22 }),
   "M01", 22, "with({day})");
 
+TemporalHelpers.assertPlainMonthDay(md.with({ month: 12 }),
+  "M12", 15, "with({month})");
+
 TemporalHelpers.assertPlainMonthDay(md.with({ monthCode: "M12" }),
   "M12", 15, "with({monthCode})");
 
-assert.throws(TypeError, () => md.with({ month: 12 }), "with({month})");
```

Comment:
This `with` would previously have been expected to throw, but no longer does in the ISO 8601 calendar.